### DISCUSSION TITLE
How to close a  menu outside it

### DISCUSSION THREAD

**Discussion by xcaptain** [dotnet/aspnetcore#54810] [Q&A] [answered]
Using the template as an example, everytime click the button would open/close the menu

https://github.com/dotnet/aspnetcore/blob/63c8031b6a6af5009b3c5bb4291fcc4c32b06b10/src/Components/WebAssembly/Samples/HostedBlazorWebassemblyApp/Client/Shared/NavMenu.razor#L33-L36

I want to ask how to close the menu when click otherwhere of the page, a general solution is to add a global event listener

```js
document.addEventListener('click', collapseTheMenu)
```

But in blazor this global event listener doesn't know the inner state of the menu component, so what's the recommended way to solve it in blazor?

**Comment by danroth27** [comment]
To share state across components you have some options:

- Pass [parameters](https://learn.microsoft.com/aspnet/core/blazor/components#component-parameters) from parent components to child components.
- Use [cascading values and parameters](https://learn.microsoft.com/aspnet/core/blazor/components/cascading-values-and-parameters).
- Use [dependency injection](https://learn.microsoft.com/aspnet/core/blazor/fundamentals/dependency-injection) to have a shared service.

**Comment by mayur-ekbote** [comment, marked as answer]
There is no direct solution in Blazor. You have only two options:
1. Implement the JS functionality upon opening the menu. It would require you to use JSInterop 
2. Show the menu on a fixed backdrop modal. Add the closing logic in the @onclick event of the backdrop

**Reply by xcaptain** [reply]
Can you be more specific about using JSinterop?

I know first step is `document.addEventListener('click', collapseTheMenu)` but how do the global event listener collapse the blazor menu component?

I found it's a little difficult to implement a dropdown menu in blazor

**Reply by xcaptain** [reply]
The fluentUI is using a fullscreen overlay layer, click elsewhere will trigger the overlay close, https://github.com/microsoft/fluentui-blazor/blob/dev/src/Core/Components/Menu/FluentMenu.razor

but I don't think it's a good solution.

**Reply by xcaptain** [reply]
I think the easiest way to implement a collapsible menu is using the popover api, but saddly this feature is not widely used.
